Source/WebKit/WebProcess/WebPage/UnifiedTextReplacementController.h

```diff
@@ -56,10 +58,21 @@ class UnifiedTextReplacementController final {
 
     void didEndTextReplacementSession(const WTF::UUID&, bool accepted);
 
+    void textReplacementSessionDidReceiveTextWithReplacementRange(const WTF::UUID&, const WebCore::AttributedString&, const WebCore::CharacterRange&, const WebKit::WebUnifiedTextReplacementContextData&);
+
+    void textReplacementSessionDidReceiveEditAction(const WTF::UUID&, WebKit::WebTextReplacementData::EditAction);
+
 private:
+    struct Replacement {
+        WebCore::AttributedString attributedText;
+        WebCore::CharacterRange range;
+    };
+
     WeakPtr<WebPage> m_webPage;
 
     HashMap<WTF::UUID, Ref<WebCore::Range>> m_contextRanges;
+    HashMap<WTF::UUID, Ref<WebCore::DocumentFragment>> m_originalDocumentNodes;
+    HashMap<WTF::UUID, Vector<Replacement>> m_replacements;
```

Comment:
Should something be clearing these at some point?

Reply:
I guess the `didEnd` delegate method can clear these